Source/player.cpp

```diff
@@ -2145,8 +2145,11 @@ void LoadPlrGFX(Player &player, player_graphic graphic)
 		szCel = "qm";
 		break;
 	case player_graphic::Death:
-		if (animWeaponId != PlayerWeaponGraphic::Unarmed)
-			return;
+		if (animWeaponId != PlayerWeaponGraphic::Unarmed) {
+			// Only one Death animation exists, for unarmed characters
+			player._pgfxnum &= ~0xFU;
+			animWeaponId = PlayerWeaponGraphic::Unarmed;
+		}

```

Comment:
Now that the assert is gone, it's become apparent to me that the only thing we need to do is update the value of `animWeaponId`. We don't need to mask `_pgfxnum` here because it was already used to assign `animWeaponId` on line 2111. And we don't need to check the value of `animWeaponId` because we're forcing it to `PlayerWeaponGraphic::Unarmed` regardless.

```suggestion
		// Only one Death animation exists, for unarmed characters
		animWeaponId = PlayerWeaponGraphic::Unarmed;
```